### ISSUE TITLE
Tx gasLimit > 100000000 causes gas limit reached error and stuck tx

### ISSUE THREAD

**Issue by dankostiuk** [0xPolygon/polygon-edge#174]
### Description

It seems that setting a tx gasLimit higher than 100000000 results in a pending tx that will never confirm. Any txs after this from the same account will also get stuck with no way of overwriting them.

https://github.com/0xPolygon/polygon-sdk/blob/335e78199b52c768e9b00c3f955e9a42a3ce5ed4/consensus/ibft/ibft.go#L356

Looking at the code above, a block header is initialized with its gasLimit set to 100000000. All tx gasLimits within that same block must not surpass this limit or else this issue occurs.

Aside from these transactions getting 'stuck', I observed many occurrences of the following log messages:
```
[ERROR] polygon: failed to apply tx: err=\"gas limit reached in the pool\""}

[31mERROR\u001b[0m\tbasichost\tfailed to resolve local interface addresses\t{\"error\": \"route ip+net: netlinkrib: too many open files\"}"}

[ERROR] polygon.network: failed to dial bootnode: err=\"peer rejected; insufficient capacity\""}
```

### Your environment

* OS and version `Ubuntu 20`
* version of the Polygon SDK `develop`
* branch that causes this issue `96be904`

### Steps to reproduce

* Tell us how to reproduce this issue <br />
* Where the issue is, if you know <br />
* Which commands triggered the issue, if any

1. Send a tx with gasLimit 5,000,000,000
2. Observe the `gas limit reached in the pool` log messages
3. Observe that subsequent txs from the same address cannot be processed
4. Attempting to override the nonce of the stuck tx (with a lower gasLimit but higher gasPrice) had no effect

### Expected behaviour

I expect there to be some sort of threshold that a tx's gasLimit must be under before it can be accepted. This threshold should be well under the hardcoded header gasLimit (aka Transition.gasPool) of 100000000.

### Actual behaviour

A tx with an extremely high gasLimit still gets picked up resulting in indefinite pending txs and error messages.

**Comment by 0xpanoramix** [comment]
Hi @dankostiuk, I was able to reproduce the issue, thanks for the information you've provided.
I'll get back to you as soon as I have made some tests.

**Comment by 0xpanoramix** [comment]
Hi @dankostiuk , I found the reason of this abnormal behaviour.

The transaction with the gas limit set to 5 000 000 000 was marked as invalid, but pushed back to the transaction pending queue afterward.
Transactions with a gas limit greater than the block one should be discarded, since they will never be able to fit in a block.

To avoid this behaviour, I've created a PR - #183 - which introduces a way to mark if any error encountered during the transaction execution is recoverable.
